GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -1883,26 +1883,30 @@ void InGameUI::update( void )
 		layout->runUpdate();
 	}
 
-	//Handle keyboard camera rotations
-	if( m_cameraRotatingLeft && !m_cameraRotatingRight )
+	if (m_cameraRotatingLeft || m_cameraRotatingRight || m_cameraZoomingIn || m_cameraZoomingOut)
 	{
-		//Keyboard rotate left
-		TheTacticalView->setAngle( TheTacticalView->getAngle() - TheGlobalData->m_keyboardCameraRotateSpeed );
-	}
-	if( m_cameraRotatingRight && !m_cameraRotatingLeft )
-	{
-		//Keyboard rotate right
-		TheTacticalView->setAngle( TheTacticalView->getAngle() + TheGlobalData->m_keyboardCameraRotateSpeed );
-	}
-	if( m_cameraZoomingIn && !m_cameraZoomingOut )
-	{
-		//Keyboard zoom in
-		TheTacticalView->zoomIn();
-	}
-	if( m_cameraZoomingOut && !m_cameraZoomingIn )
-	{
-		//Keyboard zoom out
-		TheTacticalView->zoomOut();
+		// TheSuperHackers @tweak Decouples camera rotation and zoom from draw (aka render update)
+		const Real logicTimeScaleOverFpsRatio = TheGameEngine->getActualLogicTimeScaleOverFpsRatio();
```

Comment:
I think the camera zoom and rotate should be decoupled from the logic time scale as well. It should just always be the same speed unless the user changes some specific option for it.